pkg/sql/schemachanger/scexec/dependencies.go

```diff
@@ -222,6 +222,9 @@ type IndexSpanSplitter interface {
 
 	// MaybeSplitIndexSpans will attempt to split the backfilled index span.
 	MaybeSplitIndexSpans(ctx context.Context, table catalog.TableDescriptor, indexToBackfill catalog.Index) error
+
+	// MaybeSplitIndexSpansForPartitioning will attempt to split the backfilled index span.
```

Comment:
perhaps explain how it differs from MaybeSplitIndexSpans?

Reply:
// MaybeSplitIndexSpans will attempt to split the backfilled index span, if
	// the index is in the system tenant or is partitioned.

	// MaybeSplitIndexSpansForPartitioning will split backfilled index spans
	// across hash-sharded index boundaries if applicable.